src/main/java/org/walkmod/OptionsBuilder.java

```diff
@@ -30,26 +36,49 @@
  * @author abelsromero
  */
 public class OptionsBuilder {
+	private static final String DEFAULT_PATH = ".";
 
 	/**
-	 * Options instance
+	 * Collected options
 	 */
-	private Options options;
+	private final Map<String, Object> options = new HashMap<String, Object>();
```

Comment:
The typical Immutable + Builder pairs we use are indeed typically encapsulated in the Immutable classes we use (Mostly using immutables.org library for that).
The coupling between the Builder and the Built is tight per definition
But look how the Options class can be strict in constructor now because it follows strict
deep immutables semantics.
When we are at it: I don't like the leakage of inner information done by the public constants on Options. Should they be package protected instead?
Another question: You seem to prefer "null" over empty collections/maps. For mutable data I have different opinions based on use case but for immutable data I am strict in the "empty instead of null" camp because it costs you no memory. What do you thing?